File: reddit-api-spring/src/main/java/com/gusrylmubarok/reddit/backend/repository/VoteRepository.java
package com.gusrylmubarok.reddit.backend.repository;

import com.gusrylmubarok.reddit.backend.model.Post;
import com.gusrylmubarok.reddit.backend.model.User;
import com.gusrylmubarok.reddit.backend.model.Vote;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface VoteRepository extends JpaRepository<Vote, Long> {
    Optional<Vote> findTopByPostAndUserOrderByVoteIdDesc(Post post, User currentUser);
}

File: src/main/java/com/reddot/repository/VoteRepository.kt
package com.reddot.repository

import com.reddot.entity.Post
import com.reddot.entity.User
import com.reddot.entity.Vote
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*


interface VoteRepository : JpaRepository<Vote, Long> {
    fun findTopByPostAndUserOrderByIdDesc(post: Post, currentUser: User): Optional<Vote>
}